planning/autoware_static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp

```diff
@@ -120,11 +151,39 @@ OptimizationTrajectoryBasedCenterline::generate_centerline_with_optimization(
 }
 
 std::vector<TrajectoryPoint> OptimizationTrajectoryBasedCenterline::optimize_trajectory(
-  const Path & raw_path) const
+  const PathWithLaneId & raw_path_with_lane_id, const Path & raw_path,
+  std::shared_ptr<RouteHandler> & route_handler_ptr_, LaneletMapBin::ConstSharedPtr & map_bin_ptr_,
+  const std::string & goal_method, const double & refine_goal_search_radius_range) const
 {
-  // convert to trajectory points
-  const auto raw_traj_points = [&]() {
-    const auto raw_traj = convert_to_trajectory(raw_path);
+  // create path_with_lane_id by goal_method
+  PathWithLaneId path_with_lane_id_points;
+  if (goal_method == "") {
```

Comment:
We should have a name for this goal_method since it has an explicit meaning.
How about `None`?